docs/rules/destructuring-assignment.md

```diff
@@ -0,0 +1,87 @@
+# Enforce consostent usage of destructuring assignment of props, state and context (react/destructuring-assignment)
+
+Rule can be set to either of `always`, `never`, `ignore` for `class` and `SFC` components.
```

Comment:
this should probably mention `createClass` components as well

Reply:
@ljharb, do you suggest `createClass` as additional option? So it would be [`SFC`, `createClass`, `class`]?  If so can it be done as a separate PR? Right now `class` option includes `createClass`.

Reply:
Yes, I think they should be separate options - i think it'd be fine to add it in a separate PR, but I don't think this should be merged with `class` handling both :-/

Reply:
hmmm, could you explain rationale behind that? To me this rule almost can be monolithic (cover all possible react components with a single setting). I can see the need for SFC vs Class components distinction, but not class vs createClass. What is the use case?

Reply:
For one, createClass components are often legacy, and legacy things need different rules. Mainly though, all three are actually distinct constructs - createClass components have different behavior than class-based components, in particular around auto-binding all the "instance" methods;

I think that it'd be fine to allow a string *or* an object option, where the string option is a single setting for all components - but the object option is intended to allow fine-grained control, so it's supposed to be *not* monolithic by design.

Reply:
I agree yeah, technically if we are going to list component types, we should list them all. But even though it is very little code to be added to support the 3rd `createClass` option, I am not very comfortable adding a special config option for a deprecated functionality. Additionally the  difference in behaviour between `class` and `createClass` component doesn't seem to be related to what this rule is checking (in both props are referenced with `this.props` right, or am I wrong?).  

So seems like the single string option controlling all components is an acceptable solution for both of us and is common pattern across the library. I will update my pr shortly.

Reply:
@ljharb I added an incremental change to switch to a single string option so it might be easier to review.